src/types/index.ts

```diff
@@ -15,9 +15,27 @@
 import {Level, Logger} from '../logger';
 import {Octokit} from '@octokit/rest';
 
+type FileMode = '100644' | '100755' | '040000' | '160000' | '120000';
+
+/**
+ * GitHub definition of tree
+ */
+declare interface GitCreateTreeParamsTree {
+  path?: string;
+  mode?: FileMode;
+  type?: 'blob' | 'tree' | 'commit';
+  sha?: string | null;
+  content?: string;
+}
+
+interface FileData {
+  mode: FileMode;
```

Comment:
Do we expect all change sources to include the file mode or should it be optional with a default value?

Reply:
I think it's a great idea. The only downside I can see is that we will diverge from the Octokit/GitHub v3 api design where there is no default value. However users don't need to know that we use Octokit - just our interface, so there shouldn't be a problem.